src/transforms/validcoords.jl

```diff
@@ -0,0 +1,58 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    ValidCoords(CRS)
+    ValidCoords(code)
+
+Retain the geometries within the domain of the
+projection of type `CRS` or with EPSG/ESRI `code`.
+"""
+struct ValidCoords{CRS} <: CoordinateTransform end
+
+ValidCoords(CRS) = ValidCoords{CRS}()
+
+ValidCoords(code::Type{<:EPSG}) = ValidCoords(CoordRefSystems.get(code))
+
+ValidCoords(code::Type{<:ESRI}) = ValidCoords(CoordRefSystems.get(code))
+
+parameters(::ValidCoords{CRS}) where {CRS} = (; CRS)
+
+preprocess(t::ValidCoords, d::Domain) = findall(g -> all(_isvalid(t), pointify(g)), d)
+
+function preprocess(t::ValidCoords, d::Mesh)
+  isvalid = _isvalid(t)
+  points = vertices(d)
+  topo = topology(d)
+  ∂₂₀ = Boundary{2,0}(topo)
+  findall(1:nelements(d)) do elem
+    is = ∂₂₀(elem)
+    all(isvalid(points[i]) for i in is)
+  end
+end
+
+preprocess(t::ValidCoords, d::GeometrySet{<:Manifold,<:CRS,<:Union{Polytope,MultiPolytope}}) =
+  findall(g -> all(_isvalid(t), eachvertex(g)), d)
```

Comment:
We try to avoid dispatching on multiple type parameters like this. Try to dispatch on the element type of the GeometrySet instead. You can have a fallback method for Geometry, and other methods for Polytope and MultiPolytope